src/trace/span-inferrer.ts

```diff
@@ -97,9 +104,27 @@ export class SpanInferrer {
       options.tags.connection_id = event.requestContext.connectionId;
       options.tags.event_type = event.requestContext.eventType;
     }
-    if (parentSpanContext) {
-      options.childOf = parentSpanContext;
+    let upstreamAuthorizerSpan: SpanWrapper | undefined;
+    if (this.isTracedAuthorizerInvocation(event) && event.requestContext.authorizer.integrationLatency > 0) {
+      try {
+        const parsedUpstreamContext = JSON.parse(event.requestContext.authorizer._datadog);
+        let upstreamSpanOptions: SpanOptions = {};
+        upstreamSpanOptions = {
+          startTime: parsedUpstreamContext[parentSpanFinishTimeHeader],
+          tags: { operation_name: "aws.apigateway.authorizer", ...options.tags },
+        };
+        upstreamSpanOptions.childOf = parentSpanContext;
+        upstreamAuthorizerSpan = new SpanWrapper(
+          this.traceWrapper.startSpan("aws.apigateway.authorizer", upstreamSpanOptions),
+          { isAsync: false },
+        );
+        upstreamAuthorizerSpan.finish(event.requestContext.timeEpoch);
```

Comment:
The `event.requestContext.timeEpoch` header exists for HTTP API (API Gateway V2).  
For REST API (API Gateway V1), the corresponding key is `event.requestContext.requestTimeEpoch`.
I also found out that `integrationLatency` is only available for REST API. 
Moreover, for HTTP API the key actually becomes  `event.requestContext.authorizer.lambda._datadog` [example](https://github.com/DataDog/datadog-lambda-python/blob/joey/authorizer_spans/tests/event_samples/authorizer-request-api-gateway-v2.json#L25-L26). So this current code here fits mostly with REST API. 

Also, in both cases, the epoch time here is the original user invocation timestamp. Therefore, I was using `the request time epoch + integrationlatency` to get an approximation in [the python version](https://github.com/DataDog/datadog-lambda-python/blob/joey/authorizer_spans/datadog_lambda/tracing.py#L712-L720). Please let me know what do you think.

In short, my suggestion for this line is
`upstreamAuthorizerSpan.finish(event.requestContext.requestTimeEpoch + event.requestContext.authorizer.integrationLatency);`